packages/tempo/doc/architecture.md

```
@@ -3,6 +3,27 @@
+
+### TempoRuntime — single hardened bridge (v2.2+)
+
+Prior to v2.2, Tempo spread its inter-module state across many `globalThis[Symbol.for(…)]` slots (`$terms`, `$extends`, `$modules`, `$installed`, `$reset`, `$Plugins`, `$Register`).  Each slot was a potential tamper target and the scattered writes made the global namespace harder to audit.
+
+As of v2.2, all of that bookkeeping is consolidated inside a single **`TempoRuntime`** object (`src/tempo.runtime.ts`).  The runtime is stored on `globalThis` under one hardened property:
+
+```typescript
+Symbol.for('magmacomputing/tempo/runtime')
+```
+
+The property descriptor is `enumerable: false, configurable: false, writable: false`.  External code can neither replace nor delete the runtime.
```

markdownlint-cli2 (0.22.0)

[warning] Link fragments should be valid

(MD051, link-fragments)